docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,212 @@
+    AutoRoutine routine = autoFactory.newRoutine("branching2024Auto");
+
+    // This routine uses segments between pre-defined handoff points.
+    // Expand the tooltip for information about their naming convention -> (1)
+
+    // Load the routine's trajectories
+    AutoTrajectory startToC2 = routine.trajectory("startToC2");
+    AutoTrajectory C2toM1 = routine.trajectory("C2toM1");
+    AutoTrajectory M1toM2 = routine.trajectory("M1toM2");
+    AutoTrajectory M2toM3 = routine.trajectory("M2toM3");
+    AutoTrajectory M1toScore = routine.trajectory("M1toScore");
+    AutoTrajectory M2toScore = routine.trajectory("M2toScore");
+    AutoTrajectory M3toScore = routine.trajectory("M3toScore");
+    AutoTrajectory scoreToM2 = routine.trajectory("scoreToM2");
+    AutoTrajectory scoreToM3 = routine.trajectory("scoreToM3");
+
+    // When the routine starts, shoot the first gamepiece, then go to the "C2" location
+    routine.running().onTrue(shooterSubsystem.shoot().andThen(startToC2.cmd()));
+
+    // Pick up and shoot the gamepiece at the "C2" location, then go to the "M1" location
+    startToC2.active().whileTrue(intakeSubsystem.intake());
+    startToC2.done().onTrue(shooterSubsystem.shoot().andThen(C2toM1.cmd()));
+
+    // Run the intake when we are approaching a gamepiece
+    C2toM1.active()
+        .or(scoreToM2.active())
+        .or(scoreToM3.active())
+        .or(M1toM2.active())
+        .or(M2toM3.active())
+        .whileTrue(intakeSubsystem.intake());
+
+    // If we picked up the gamepiece, go to the "Score" location
+    // If we didn't pick up the gamepiece, go to the next midline location
+    C2toM1.done().onTrue(either(M1toScore.cmd(), M1toM2.cmd(), shooterSubsystem::hasGamepiece));
+    scoreToM2.done().or(M1toM2.done()).onTrue(either(M2toScore.cmd(), M2toM3.cmd(), shooterSubsystem::hasGamepiece));
+    scoreToM3.done().or(M2toM3.done()).onTrue(either(M3toScore.cmd(), Commands.none(), shooterSubsystem::hasGamepiece));
+
+    // After we go to score, shoot then go to the next available midline location
+    M1toScore.done().onTrue(shooterSubsystem.shoot().andThen(scoreToM2.cmd()));
+    M2toScore.done().onTrue(shooterSubsystem.shoot().andThen(scoreToM3.cmd()));
+    M3toScore.done().onTrue(shooterSubsystem.shoot());
+
+    return routine;
 }
 ```
 
+1. ![Samples view layer showing ](../media/choreolib-branching-auto.png)
+
+## AutoChooser
+
+The `AutoChooser` ([Java](/api/choreolib/java/choreo/auto/AutoChooser.html)) class allows you to send a list of your autonomous routines to a driver dashboard for selection before a match. It is meant to be a more efficient alternative to `SendableChooser`, taking a [lazy loading](https://en.wikipedia.org/wiki/Lazy_loading) approach to generating command compositions or an `AutoRoutine`. This approach has the benefit of not loading all autos on startup, but also not loading the auto after the match starts, which may cause a delay when using many or large trajectories.
```

Comment:
```suggestion
The `AutoChooser` ([Java](/api/choreolib/java/choreo/auto/AutoChooser.html)) class allows you to send a list of your autonomous routines to a driver dashboard for selection before a match. It is meant to be a more efficient alternative to `SendableChooser`, taking a [lazy loading](https://en.wikipedia.org/wiki/Lazy_loading) approach to generating command compositions or an `AutoRoutine`. This approach has the benefit of not loading all autos on startup and not causing your robot to go into a constant crash loop if you have a typo in an auto, but also not loading the auto after the match starts, which may cause a delay when using many or large trajectories.
```

Reply:
It doesn't prevent crashes it prevents crashes that stop your code until redeploy. If it loads on init then the robot will crash and you can't ever recover from that state. AutoChooser prevents that as when the robot restarts you just pick a different auto

Reply:
That makes sense, I'll try to fit it in but if I can't word it in a nice way I may just omit it, as imho it isn't really an important feature of the API. Boot looping doesn't really present a huge problem in FRC; just pause the console, fix your code, and redeploy.